File: prng-linear-congruential-generator/prng-linear-congruential-generator.js
class LCG {
  constructor(x){
    this.a=2;
    this.c=3;
    this.m=10;
    this.x=x;
  }
  random() {
    let d =(this.a*this.x+this.c)%this.m
    this.x = d
    return d/10
  }
}

File: prng-linear-congruential-generator/prng-linear-congruential-generator.py
import math
class LCG(object):
  a = 2
  c = 3
  m = 10
  def __init__(self, seed):
      self.seed = seed
  def random(self):
      self.seed = (self.a*self.seed+self.c)%self.m
      return self.seed/10 